lib/internal/webstreams/readablestream.js

```diff
@@ -2440,7 +2441,7 @@ function setupReadableStreamDefaultController(
   const startResult = startAlgorithm();
 
   PromisePrototypeThen(
-    PromiseResolve(startResult),
+    new Promise((r) => r(startResult)),
```

Comment:
Why this change?

Reply:
to align with the WHATWG spec, we need to do it in more places too Ref: https://github.com/nodejs/node/pull/50126#issuecomment-1824957375